src/test/base64_tests.cpp

```diff
@@ -37,10 +37,21 @@ BOOST_AUTO_TEST_CASE(base64_testvectors)
     }
 
     // Decoding strings with embedded NUL characters should fail
-    BOOST_CHECK(!DecodeBase64("invalid\0"s));
-    BOOST_CHECK(DecodeBase64("nQB/pZw="s));
-    BOOST_CHECK(!DecodeBase64("nQB/pZw=\0invalid"s));
-    BOOST_CHECK(!DecodeBase64("nQB/pZw=invalid\0"s));
+    BOOST_CHECK(!DecodeBase64("invalid\0"s)); // correct size, invalid due to \0
+    BOOST_CHECK( DecodeBase64("nQB/pZw="s)); // valid
+    BOOST_CHECK(!DecodeBase64("nQB/pZw=invalid"s)); // invalid, padding only allowed at the end
```

Comment:
Last line no longer has an embedded NUL character as specified in the comment above this code block (`// Decoding strings with embedded NUL characters should fail`). Might belong more in `base64_padding` or at least in its own newline-separated block here:
```suggestion
    BOOST_CHECK(!DecodeBase64("invalid\0"s)); // correct size, invalid due to \0
    BOOST_CHECK( DecodeBase64("nQB/pZw="s)); // valid

    BOOST_CHECK(!DecodeBase64("nQB/pZw=invalid"s)); // invalid, padding only allowed at the end
```

Reply:
moved the values without \0 above the comment in 32 and 64 tests, thanks!